pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/AggregationOperator.java

```diff
@@ -97,15 +69,12 @@ public Block getNextBlock(BlockId blockId) {
             _numTotalRawDocs);
 
     // Build intermediate result block based on aggregation result from the executor.
-    List<Serializable> aggregationResults = _aggregationExecutor.getResult();
-    return new IntermediateResultsBlock(AggregationFunctionFactory.getAggregationFunction(_aggregationInfoList),
-        aggregationResults);
+    return new IntermediateResultsBlock(_aggregationFunctionContexts, aggregationExecutor.getResult(), false);
   }
 
   @Override
-  public boolean open() {
-    _projectionOperator.open();
-    return true;
+  public Block getNextBlock(BlockId blockId) {
```

Comment:
Actually we never allow calling getNextBlock with a blockId (the only exception is DataSource). Operators that allow calling getNextBlock() multiple times are BReusableFilterDocIdSetOperator and MProjectionOperator, and both of them will throw exception when getting called with a blockId. So to get symmetric behavior across all operators, the easier way is never call getNextBlock with a BlockId.